core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java

```diff
@@ -222,7 +222,7 @@ public String getAddress() {
 
                 @Override
                 public String getGroup() {
-                  return entry.getResourceGroup();
+                  return entry.getResourceGroup().canonical();
```

Comment:
was the return type of this method not changed because its SPI?  We can change SPI w/o deprecation cycle if we want to.  Using a narrow type instead of string is so much nicer.

Reply:
Fixed this and other places in updates after your review. I was initially being cautious about changing anything client facing. I went back and did another pass today.